test/echo_tests.go

```diff
@@ -0,0 +1,17 @@
+package test
+
+import (
+	"testing"
+)
+
```

Comment:
Your tests ideally need to cover more usage scenarios, for example
1. https/http2 senario
2. a senario that the echo server has already used a middleware, to verify that if your code rewrite the middleware
3. exception senario